src/Microsoft.ComponentDetection.Detectors/conda/CondaDependencyResolver.cs

```diff
@@ -0,0 +1,110 @@
+    /// <param name="dependencyPlatform">The platform of the dependency that should expanded.</param>
+    /// <param name="parentId">The id of the parent component of the dependency that should expanded.</param>
+    /// <param name="condaLock">The full condaLock object.</param>
+    /// <param name="singleFileComponentRecorder">The SingleFileComponentRecorder.</param>
+    public static void RecordTransitiveDependencies(string dependencyName, string dependencyPlatform, string parentId, CondaLock condaLock, ISingleFileComponentRecorder singleFileComponentRecorder)
+    {
+        condaLock.Package.ForEach(package =>
+        {
+            if (package.Name == dependencyName && package.Platform == dependencyPlatform)
+            {
+                RecordPackage(package, false, dependencyPlatform, parentId, condaLock, singleFileComponentRecorder);
+            }
+        });
+    }
+
+    /// <summary>
+    /// Registers the usage of a package.
+    /// </summary>
+    /// <param name="package">The conda package to register.</param>
+    /// <param name="isExplicitDependency">True if the package is explicitly referenced.</param>
+    /// <param name="dependencyPlatform">The platform of the dependency that should expanded.</param>
+    /// <param name="parentId">The id of the parent component of the dependency that should expanded.</param>
+    /// <param name="condaLock">The full condaLock object.</param>
+    /// <param name="singleFileComponentRecorder">The SingleFileComponentRecorder.</param>
+    public static void RecordPackage(CondaPackage package, bool isExplicitDependency, string dependencyPlatform, string parentId, CondaLock condaLock, ISingleFileComponentRecorder singleFileComponentRecorder)
+    {
+        var condaLockComponent = CreateCondaLockComponentFromPackage(package);
+        var pipComponent = CreatePipComponentFromPackage(package);
+        var componentId = package.Manager == "pip" ? pipComponent.Id : condaLockComponent.Id;
+
+        singleFileComponentRecorder.RegisterUsage(
+            new DetectedComponent(package.Manager == "pip" ? pipComponent : condaLockComponent),
+            isExplicitReferencedDependency: isExplicitDependency,
+            parentComponentId: parentId);
+
+        package.Dependencies.Keys.ToList().ForEach(dependency =>
+        {
+            RecordTransitiveDependencies(dependency, dependencyPlatform, componentId, condaLock, singleFileComponentRecorder);
+        });
+    }
+
+    /// <summary>
+    /// Converts a CondaPackage to a CondaLockComponent.
+    /// </summary>
+    /// <param name="package">The CondaPackage to convert.</param>
+    /// <returns>The CondaLockComponent.</returns>
+    public static CondaLockComponent CreateCondaLockComponentFromPackage(CondaPackage package)
+        => new CondaLockComponent(
+            package.Name,
+            package.Version,
+            package.Category,
+            package.Dependencies,
```

Comment:
As mentioned in other file, dependencies are not reported directly as part of the component. The graph is generated by the `SingleFileComponentRecorder`, I would only expect you do loop over the dependencies property and register them individually with current component as parent ID

Reply:
As discussed in the other file, the code has been refactored accordingly. :)